src/Services/Implementations/DefaultAlgoliaInsightsService.cs

```diff
@@ -78,10 +79,12 @@ private uint Position
         /// <summary>
         /// Initializes a new instance of the <see cref="DefaultAlgoliaInsightsService"/> class.
         /// </summary>
-        public DefaultAlgoliaInsightsService(IAlgoliaRegistrationService algoliaRegistrationService, IInsightsClient insightsClient)
```

Comment:
This is a breakinch change. I don't see a reason to keep this and other service implementations as public classes. There is no extension point + Custom service still can call the original implementation thru decorator, see https://docs.xperience.io/custom-development/decorating-system-services#Decoratingsystemservices-Decorateservicesviadependencyinjection

Reply:
That makes sense, but after marking them `internal` they can no longer be referenced by the test project, e.g. https://github.com/Kentico/xperience-algolia/blob/master/tests/Tests/IAlgoliaIndexingServiceTests.cs#L33. Am I doing something wrong there?

Reply:
I believe you can use [InternalsVisibleTo](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-6.0) to expose the `internal` types to the unit test project.